### ISSUE TITLE
"minikube ip" address does not match with "minikube service <service-name>"

### ISSUE THREAD

**Issue by Carlosgpg1** [kubernetes/minikube#10580]
I use minikube in Windows 10, and start with "minikube start", using docker as Hypervisor. 

```
C:\Users\carlo\serviciosCursoKub>minikube start
* minikube v1.17.1 en Microsoft Windows 10 Home 10.0.19041 Build 19041
* Using the docker driver based on existing profile
* Starting control plane node minikube in cluster minikube
* Restarting existing docker container for "minikube" ...
! StartHost failed, but will try again: provision: get ssh host-port: get port 22 for "minikube": docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" minikube: exit status 1
stdout:


stderr:
Template parsing error: template: :1:4: executing "" at <index (index .NetworkSettings.Ports "22/tcp") 0>: error calling index: index of untyped nil

* Restarting existing docker container for "minikube" ...
* Preparando Kubernetes v1.20.2 en Docker 20.10.2...
* Verifying Kubernetes components...
* Enabled addons: storage-provisioner, default-storageclass, dashboard
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

Okey. When i use "minikube ip" i get the addres 192.168.49.2. And when i use "minikube service --url <service-name>" i get 127.0.0.1. And i don't know why.

with "minikube ssh" i can see:

![ip](https://user-images.githubusercontent.com/47037827/108876639-bbedd100-75fe-11eb-9045-35670dad3da3.PNG)

Thanks!

**Comment by priyawadhwa** [comment]
Hey @Carlosgpg1 thanks for opening this issue. So, with the docker driver, I think `minikube ip` always returns 127.0.01 -- this was a design decision made when implementing the driver (some details can be found in this comment and this thread in general https://github.com/kubernetes/minikube/issues/7344#issuecomment-607912777).

Are you looking to forward a LoadBalancer service?

**Comment by abhimjoshi** [comment]
Hey @priyawadhwa I have resolved the issue for the discussion I was involved at the thread mentioned below: 
#7344  [(See my comment)](https://github.com/kubernetes/minikube/issues/7344#issuecomment-791900105)

It was basically a container port issue.

**Comment by medyagh** [comment]
@abhimjoshi @Carlosgpg1 the direct IP of docker containers on windows is not available, and minikube has to use a tunnel over 127.0.0.1 

for external IP address you could use `minikube tunnel`
that would require keeping the tunnel open in another window
